tests/unittests/test_feature_events_primary_keys.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+from tap_pendo.streams import FeatureEvents
+
+class TestFeatureEventsPrimaryKeys(unittest.TestCase):
+
+    def test_feature_event_primary_key_with_hourRange(self):
+        '''
+            Verify that primary keys should have expected fields with 'hour' field when period is hourRange 
+        '''
+        config = {"period": "hourRange"} # set hourRange as a period
+        expected_primary_keys = ["feature_id", "visitor_id", "account_id", "server", "remote_ip", "user_agent", "hour"]
+        feature_event_stream1 = FeatureEvents(config) # Initialize FeatuereEvents object which sets primary keys
+
+        self.assertEqual(feature_event_stream1.key_properties, expected_primary_keys)
+
+        # Reset key properties for other test as it's class variable
+        FeatureEvents.key_properties = ['feature_id', 'visitor_id', 'account_id', 'server', 'remote_ip', 'user_agent']
```

Comment:
@savan-chovatiya Test cases should not be dependent on each other.
Please include this statement at the starting of the next test case which requires this class variable in the reset mode

Reply:
I removed that comment and initialize key_properties with the default value at the start of the test.